packages/core/contracts/financial-templates/perpetual-multiparty/PositionManager.sol

```diff
@@ -808,16 +775,12 @@ contract PositionManager is FeePayer, AdministrateeInterface {
     // These internal functions are supposed to act identically to modifiers, but re-used modifiers
     // unnecessarily increase contract bytecode size.
     // source: https://blog.polymath.network/solidity-tips-and-tricks-to-save-gas-and-reduce-bytecode-size-c44580b218e6
-    function _onlyOpenState() internal view {
-        require(contractState == ContractState.Open, "Contract state is not OPEN");
-    }
-
-    function _onlyPreExpiration() internal view {
-        require(getCurrentTime() < expirationTimestamp, "Only callable pre-expiry");
+    function _notEmergencyShutdown() internal view {
+        require(emergencyShutdownTimestamp == 0, "Contract emergency shutdown");
     }
 
-    function _onlyPostExpiration() internal view {
-        require(getCurrentTime() >= expirationTimestamp, "Only callable post-expiry");
+    function _onlyEmergencyShutdown() internal view {
```

Comment:
I feel like these 2 func's can be collapsed into one: `_isEmergencyShutdown`. Then you can create 2 modifiers: `notEmergencyShutdown` and `isEmergencyShutdown`.

Reply:
ye, For sure. Am happy to join them together. would also make the logic for https://github.com/UMAprotocol/protocol/pull/2072/files#r501043517 cleaner

Reply:
In fact, this is not really possible to do cleanly, unfortunately. The way we do modifiers by using internal functions to act as logic checks then calling the function from the modifier means that the `require` statements needs to occur in the internal function (not the modifier) and as such I cant re-use this logic. To re-use the logic we'd need to put the require within the modifier, not the internal function.

I tried re-factoring this by removing these internal functions and putting everything in the requires as this seams cleaner. However, this uses measurably more bytecode. with logic in internal functions and then called by requires the contract bytecode size is 12657. After doing this refactor it is 12905. That's not a huge gain but it's big enough to justify leaving it in the more verbose but efficient logic (as currently in the PR).

Reply:
that is interesting, i always wondered why you had these modifiers vs just doing a require in the function body. I guess its a bytecode optimization.

Reply:
Ah understood.